app/src/users/admin.py

```diff
@@ -26,4 +32,146 @@ class OrganizationAdmin(admin.ModelAdmin):
+    parameter_name = 'room_name'
+
+    def lookups(self, request, model_admin):
+        return (('No room', 'No room'),) + tuple((r.name, r.name) for r in sorted(Room.objects.only('name').distinct(), key=Room.name_to_key_orderable))
+
+    def queryset(self, request, queryset):
+        value = self.value()
+        if value == 'No room':
+            return queryset.filter(_room_name=[None])
+        elif value:
+            return queryset.filter(_room_name__contains=[value])
+
+
+class ShirtPropertiesListFilter(admin.SimpleListFilter):
+    title = 'shirt properties'
+    parameter_name = 'shirt_properties'
+
+    def lookups(self, request, model_admin):
+        return [(f'{size}-{type_}', f'{size}-{type_}') for type_, _t in SHIRT_TYPES_CHOICES
+                for size, _s in SHIRT_SIZE_CHOICES]
+
+    def queryset(self, request, queryset):
+        value = self.value()
+        if value:
+            shirt_size, shirt_type = value.split('-')
+            return queryset.filter(preferences__shirt_size=shirt_size, preferences__shirt_type=shirt_type)
+
+
+class RoomInline(admin.TabularInline):
+    model = UserRoom
+
+
+class UserPreferencesInline(admin.TabularInline):
+    model = UserPreferences
+    fk_name = 'user'
+    extra = 1
+
+
+@admin.register(UserFilters)
+class UserFiltersAdmin(admin.ModelAdmin):
+    list_display = ('first_name', 'last_name', 'room_name', 'payment_accepted', 'have_preferences', 'have_lecture',
+                    'shirt_properties')
+    list_display_links = ('first_name', 'last_name')
+    fields = ('first_name', 'last_name', 'room_name', 'payment_accepted', 'have_preferences', 'have_lecture',
+              'shirt_properties')
+    readonly_fields = ('room_name', 'payment_accepted', 'have_preferences', 'have_lecture', 'shirt_properties')
+    search_fields = ('first_name', 'last_name', '_room_name')
+    list_filter = (PaymentAcceptedListFilter, HaveLectureListFilter, HavePreferencesListFilter,
+                   ShirtPropertiesListFilter, RoomNameListFilter)
+    inlines = (UserPreferencesInline, RoomInline)
+
+    def get_queryset(self, request):
+        return super().get_queryset(request).annotate(
+            _have_preferences=Exists(UserPreferences.objects.filter(user=OuterRef('id')).only('user')),
+            _room_name=ArrayAgg('room_of_user__name'),
+            _payment_accepted=ArrayAgg('preferences__payment_accepted'),
+            _shirt_properties=ArrayAgg(Concat('preferences__shirt_size', Value('-'), 'preferences__shirt_type')),
+            _have_lectures=Exists(Lecture.objects.filter(author=OuterRef('id')).only('author')),
+        )
+
+    def shirt_properties(self, obj):
+        return obj._shirt_properties
+    shirt_properties.admin_order_field = '_shirt_properties'
+
+    def room_name(self, obj):
+        return obj._room_name
+        # Implementation have M2M Fields, only first record of room_of_user is taken in list_csv_room_by_member, what if many rooms defined?
```

Comment:
FYI: I think it shouldn't be a problem as in middle table `UserRoom` users are related by `OneToOneField` :)

Reply:
Oh, indeed. Then members=ManyToManyField is confusing (because in reality it's not M2M). I suggest removing it, because associative table does what we need.